components/count-badge/count-badge-icon.js

```diff
@@ -33,23 +33,35 @@ class CountBadgeIcon extends CountBadgeMixin(LitElement) {
 			display: inline-block;
 		}
 
+		.d2l-count-badge-number-wrapper {
+			position: absolute;
+			width: max-content;
+		}
+
 		:host {
-			padding-right: var(--d2l-count-badge-icon-padding);
+			display: inline-block;
+			position: relative;
+			padding-left: 0.5rem;
+			padding-right: var(--d2l-count-badge-icon-horizontal-padding);
 		}
 
 		:host([dir="rtl"]) {
-			padding-left: var(--d2l-count-badge-icon-padding);
-			padding-right: 0;
+			padding-left: var(--d2l-count-badge-icon-horizontal-padding);
+			padding-right: 0.5rem;
```

Comment:
Our "best-effort" padding to prevent most overflows is 0.5rem on each side, to keep it symmetrical. The default of --d2l-count-badge-icon-horizontal-padding is also 0.5rem;